### PR TITLE
feat: add --gpus flag to ilab serve

### PR DESCRIPTION
**Pull request by alimaredia**
**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/serve.py [hunk 1]
```diff
@@ -94,6 +94,11 @@ def signal_handler(
         "Automatically detected based on the model file properties.\n"
     ),
 )
+@click.option(
+    "--gpus",
+    type=click.IntRange(min=0),
+    help="Number of GPUs to utilize for evaluation (not applicable to llama-cpp backend)",
```
Comment (danmcp) [inline src/instructlab/model/serve.py:100]: utilize for serving?

File: src/instructlab/model/serve.py [hunk 2]
```diff
@@ -158,6 +164,17 @@ def serve(
         # Warn if unsupported backend parameters are passed
         warn_for_unsuported_backend_param(ctx)
 
+        if gpus:
+            # if `--tensor-parallel-size is included in ctx.args (click arguments after "--"),
+            # that value has precedence over the value of the --gpus flag.
+            if "--tensor-parallel-size" not in ctx.args:
```
Comment (danmcp) [inline src/instructlab/model/serve.py:170]: Should use contains_argument from vllm.  Generate does this now and evaluate should switch to use it at some point.

**Review by danmcp — CHANGES_REQUESTED** [review]

**Comment by cdoern** [comment]
This needs to be in the config if it's going to be a flag. I currently have a pr open for this.

**Comment by cdoern** [comment]
I think #1931 is ready and should go in since it includes the config change. The flag in that one was already called `--gpus`

**Comment by alimaredia** [comment]
Closing in favor of 1931. The changes needed to add gpus as a config are less invasive than I thought they would be.

**Pull request closed without merging**